bootstrap.sh

```diff
@@ -17,7 +17,11 @@ export APP_ROOT=$(pwd)
 export WORKSPACE=${WORKSPACE:-$APP_ROOT}  # if running in jenkins, use the build's workspace
 export BONFIRE_ROOT=${WORKSPACE}/.bonfire
 export CICD_ROOT=${BONFIRE_ROOT}
-export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+
+if [[ ! -n "$IMAGE_TAG" ]]; then
+    export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+fi
+
```

Comment:
Hi ! following the discussion with @bsquizz  and @gburges  last week, and the following debugging session me an Gabor made tests with the Jenkinsfile pipeline, and it is possible to perform a checkout to a directory (instead of mixing up the files in the current dir) by wrapping the checkout action in a `dir` block . 

Then, if you use the same `dir` block in any subsequent stage, the `git ref` you get from your bootstrap script for the IMAGE_TAG should be the correct one (not the cicd tools one, but the one from the cloned repository). 

Hence, it should not be required to manipulate the script to disable setting the IMAGE_TAG variable if already set. 

The example pipeline I tested this is:

```
pipeline {
    agent any

    stages {
        stage('Hello') {
            steps {
                dir('example-repository') {
                    git branch: 'main', url: 'https://github.com/Victoremepunto/example-repository'
                }
            }
        }
        
        stage('Second Stage') {
            steps {
                dir('example-repository') {
                    sh 'git rev-parse --short=7 HEAD'
                     // This yields the HEAD from the cloned repo, not the pipeline's repo.
                }
            }
        }
    }
    post {
        always {
            deleteDir()
        }
    }
}
```

Reply:
So that I can properly test this, can I update this in a followup PR? This is already a feature branch being used by a handful of folks.

Reply:
Let's try and see if this works, I'd like to avoid touching the bootstrap.sh script's logic if possible. A short update and revert IF the suggested solution works for me as well.